railties/lib/rails/application.rb

```diff
@@ -580,6 +584,26 @@ def validate_secret_key_base(secret_key_base)
       end
     end
 
+    def eager_load!
+      EAGER_LOAD_MUTEX.synchronize do
+        return if @eager_loaded
+
+        ActiveSupport.run_load_hooks(:before_eager_load, self)
+
+        # Checks defined?(Zeitwerk) instead of zeitwerk_enabled? because we want
+        # to eager load any dependency managed by Zeitwerk regardless of the
+        # autoloading mode of the application.
+        Zeitwerk::Loader.eager_load_all if defined?(Zeitwerk)
+
+        super unless Rails.autoloaders.zeitwerk_enabled?
+        config.eager_load_namespaces.each do |namespace|
+          namespace.eager_load! unless namespace == self
```

Comment:
```suggestion
          namespace.eager_load! unless namespace == self.class
```

This breaks otherwise with a `#<NoMethodError: protected method `eager_load!' called for #<Shopify::Application:0x00007fd188a72658>>`

Reply:
Yes, this was untested, written in the editor to be able to discuss with actual code.